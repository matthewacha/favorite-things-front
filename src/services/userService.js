import Api from './api';

export default {
  async postUser(params) {
    const data = await Api().post('/v1/user/', params);
    return data;
  },
};
