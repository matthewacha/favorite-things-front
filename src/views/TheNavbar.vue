<template>
  <div id="nav">
    <div id="logoDiv">
      <img id="imageIcon" alt="appIcon" src="../assets/appIcon.svg">
    </div>
    <div :class="notification || 'notification'">
      <v-app id="inspire">
          <v-alert id="alert-pane" :value="true" :color="color">{{message}}</v-alert>
      </v-app>
    </div>
    <!-- <router-link to="/">Home</router-link> |
    <router-link to="/about">About</router-link>-->
  </div>
</template>

<script>
import { serverBus } from "@/main";

export default {
    name: 'navbar',
    props: () => ({
        notification: String,
        color: String,
        message: String
    }),
    beforeUpdate(){
        console.log('thisone', this)
    },
    data: () => ({
    error: "",
    message: this.message,
    value: false,
    // notification: 'notification-active',
    // color: ''
  }),
}
</script>
<style lang="scss" scoped>
#alert-pane {
    margin: 0;
    z-index: 2;
}
.notification {
    display: none;
    &-active{
        margin: 0 auto;
        width: 10%;
        flex: 1 1 100;
        position: relative;
        top: 6px;
    }
}
</style>


