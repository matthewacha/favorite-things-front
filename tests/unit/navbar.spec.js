import Vue from "vue";
import { shallowMount, mount } from "@vue/test-utils";
import sinon from "sinon";
import Vuex from "vuex";
import userService from "../../src/services/userService";
import TheNavbar from "@/views/TheNavbar.vue";

Vue.use(Vuex);

describe("Login.vue", () => {
  it("renders properly", () => {
    const wrapper = shallowMount(TheNavbar);
    expect(wrapper.element).toMatchSnapshot();
  });
});

// describe("test it works", () => {
//   let actions;
//   let store;

//   beforeEach(() => {
//     actions = {
//       user: sinon.stub()
//     };
//     store = new Vuex.Store({
//       state: {},
//       actions
//     });
//   });

//   it("loads the navbar component", () => {
//     const wrapper = shallowMount(TheNavbar);
//     console.log(wrapper);
//   });
// });
