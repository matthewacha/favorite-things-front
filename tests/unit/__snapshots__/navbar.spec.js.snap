// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login.vue renders properly 1`] = `
<div
  id="nav"
>
  <div
    id="logoDiv"
  >
    <img
      alt="appIcon"
      id="imageIcon"
      src="../assets/appIcon.svg"
    />
  </div>
   
  <div />
</div>
`;
